File: rental-drive/src/services/OrderService.js
import axios from "axios";

const API_BASE_URL="http://localhost:8086/api/order";

class OrderService{
    getAllOrders=()=>{
        return axios.get(`${API_BASE_URL}/admin/all`);
    };

    confirmOrder=(id)=>{
        return axios.put(`${API_BASE_URL}/admin/confirm/${id}`);
    };

    cancelOrder=(id)=>{
        return axios.put(`${API_BASE_URL}/admin/cancel/${id}`);
    };

    deleteOrder=(id)=>{
        return axios.delete(`${API_BASE_URL}/admin/delete/${id}`);
    };


    //user

    userHistory(id){
        return axios.get(`${API_BASE_URL}/userHistory/id/${id}`);
    }

    createOrder(token){
        return axios.post(`${API_BASE_URL}/add`, {
            headers: {
                Authorization: `${token}`
            }
        })
    };
    addOrder(id, customer){
        return axios.post(`${API_BASE_URL}/add/id/${id}`, customer);
    };

    findOrderById(id){
        return axios.get(`${API_BASE_URL}/${id}`);
    };

    findOrderByVehicule(id){
        return axios.get(`${API_BASE_URL}/all/vehicule/${id}`);
    }



}
export default new OrderService;

File: Backend/src/main/java/com/car/service/OrderService.java
package com.car.service;

import java.time.LocalDateTime;
import java.util.List;

import com.car.exception.ExceptionClass;
import com.car.exception.UserException;
import com.car.exception.VehiculeException;
import com.car.model.Customer;
import com.car.model.Order;
import com.car.model.UserEntity;

public interface OrderService {
	
	//for user
	public Order createOrder(UserEntity user, Customer customer) throws UserException;
	public Order findOrderById(int id) throws ExceptionClass;
	public Order cancelOrder(int id) throws ExceptionClass;
	public List<Order> userOrderHistory(int userId) throws UserException;
	
	//for admin 
	public void deleteOrder(int id) throws ExceptionClass;
	public Order confirmOrder(int id) throws ExceptionClass;
	public List<Order> getAllOrders();
	public List<Order> findByvehicule(Long id) throws VehiculeException;
}
